### ISSUE TITLE
[Desktop] Insecure form submissions are not blocked on HTTPS pages

### ISSUE THREAD

**Issue by PoorPocketsMcNewHold** [brave/brave-browser#9932]
> Have you searched for similar issues? Before submitting this issue, please check the open issues and add a note before logging a new issue. 

The closest I could find is Issue #5115, which reference MixedContent like this, however, this one just talk about third-party calls which can, and should be enforced.

## Description 

Insecure password fields (Using HTTP calls) are not enforced or blocked by Brave.
## Steps to Reproduce

1. Consult this test website : https://www.bennish.net/mixed-content.html
_-  Notice that all the other test elements of the website are correctly blocked by Brave, due to those elements, unable to be enforced in HTTPS._
2. Enter a text in the password field (First test element on the website).
3. Click enter. The typed text from the password field should be displayed to you on http://unencrypted.kennish.net/mixed-content-submit.php (Redirected automatically).

## Actual result:

The typed text from the password field is displayed to you on http://unencrypted.kennish.net/mixed-content-submit.php (Redirected automatically).

## Expected result:
The typed text would not be sent to any destinations using HTTP.

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

`Brave | 1.9.72 Chromium: 81.0.4044.138 (Official Build) unknown (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | Linux
JavaScript | V8 8.1.307.32
Flash | (désactivated)
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36
Command Line | /usr/lib/brave-bin/brave --no-sandbox --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=WebUIDarkMode,PasswordImport,PassiveMixedContentWarning,AutoupgradeMixedContent,DnsOverHttps,MixedContentSiteSetting,SimplifyHttpsIndicator --disable-features=AutofillServerCommunication,TextFragmentAnchor,AllowPopupsDuringPageUnload,SmsReceiver,AudioServiceOutOfProcess,NotificationTriggers,LookalikeUrlNavigationSuggestionsUI,VideoPlaybackQuality --flag-switches-begin --load-media-router-component-extension=1 --enable-features=WebUIDarkMode,PasswordImport,PassiveMixedContentWarning,AutoupgradeMixedContent,DnsOverHttps,MixedContentSiteSetting,SimplifyHttpsIndicator,HeavyAdIntervention,HeavyAdPrivacyMitigations,RawClipboard --flag-switches-end --disable-webrtc-apm-in-audio-service --disable-sync
Executable Path | /usr/lib/brave-bin/brave
Profile Path | /home/pm/.config/BraveSoftware/Brave-Browser/Default`

And yes, I am aware that Brave is launched in --no-sandbox, This is an issue with the current kernel i am using, which enforce security measures, and so, force either chrome-sandbox to be chmod 4755 or the program to be launched in --no-sandbox mode.
## Version/Channel Information:

Doesn´t seems to be an actual bug of some kind, that could act differently between each release channels.
- Can you reproduce this issue with the current release? Yes.
- Can you reproduce this issue with the beta channel? Not tested.
- Can you reproduce this issue with the dev channel? Not tested.
- Can you reproduce this issue with the nightly channel? Not tested.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No. Desactivating the HTTPS enforcing setting doesn´t cause other website elements to be unblocked nonetheless.
- Does the issue resolve itself when disabling Brave Rewards? No.
- Is the issue reproducible on the latest version of Chrome? Not tested.

## Miscellaneous Information:

After reading a bit on the website linked sources on MixedContent, I´ve stumbled upon this old blog post from mozilla about password sent trought HTTP calls https://blog.mozilla.org/tanvi/2016/01/28/no-more-passwords-over-http-please/ and tought that, this comportment could actually be voluntary as a compatiblity feature.
Nonetheless, I feel this still make it for an unsecure comportment.

**Comment by dentistformyeye** [comment]
Same website, but different issue that might not be Brave's fault:
Interestingly, when I turn on EASE for the HTTPS Everywhere extension (first picture), the Freckle Face font loads successfully (bad), but when I turn EASE off (second picture), the custom font doesn't load. (I normally turn on EASE; otherwise, the extension would be redundant for Brave.)
![With EASE](https://user-images.githubusercontent.com/51429646/83055518-f1d73800-a021-11ea-9e21-3e92657309d6.png)
![Without EASE](https://user-images.githubusercontent.com/51429646/83055881-77f37e80-a022-11ea-8f87-85962ced2e17.png)

**Comment by PoorPocketsMcNewHold** [comment]
> Same website, but different issue that might not be Brave's fault:
> Interestingly, when I turn on EASE for the HTTPS Everywhere extension (first picture), the Freckle Face font loads successfully (bad), but when I turn EASE off (second picture), the custom font doesn't load. (I normally turn on EASE; otherwise, the extension would be redundant for Brave.)
> ![With EASE](https://user-images.githubusercontent.com/51429646/83055518-f1d73800-a021-11ea-9e21-3e92657309d6.png)
> ![Without EASE](https://user-images.githubusercontent.com/51429646/83055881-77f37e80-a022-11ea-8f87-85962ced2e17.png)
Interesting, because even manually desactivating Brave HTTPS function, I´m unable to make it load voluntary.

**Comment by rebron** [comment]
cc: @fmarier When you get a chance, can you assist with triaging this one?

**Comment by fmarier** [comment]
Here's a comparison of Brave v. Chrome v. Firefox.

### URL Bar
Brave:
![Screenshot from 2020-07-10 11-04-10](https://user-images.githubusercontent.com/167821/87184626-34389980-c29d-11ea-8e03-3a5f377418f7.png)
Chrome:
![Screenshot from 2020-07-10 11-07-19](https://user-images.githubusercontent.com/167821/87184817-8aa5d800-c29d-11ea-861e-654bd9bb608c.png)
Firefox:
![Screenshot from 2020-07-10 11-03-58](https://user-images.githubusercontent.com/167821/87184775-7b268f00-c29d-11ea-9374-7a0ba29b1bdb.png)
### Password field
Brave:
![Screenshot from 2020-07-10 11-07-59](https://user-images.githubusercontent.com/167821/87184941-bde86700-c29d-11ea-95a5-91b014aee247.png)
Chrome:
![Screenshot from 2020-07-10 11-08-09](https://user-images.githubusercontent.com/167821/87184950-c2ad1b00-c29d-11ea-986b-ac1abd52bf92.png)
Firefox:
![Screenshot from 2020-07-10 11-08-20](https://user-images.githubusercontent.com/167821/87185069-fc7e2180-c29d-11ea-96f4-c2c012bc94f9.png)
###  Warning dialog
Brave:
N/A
Chrome:
N/A
Firefox:
![Screenshot from 2020-07-10 11-11-21](https://user-images.githubusercontent.com/167821/87185139-191a5980-c29e-11ea-909f-5022c20ecbc2.png)

**Comment by fmarier** [comment]
This issue is tracked upstream at https://bugs.chromium.org/p/chromium/issues/detail?id=897374

**Comment by fmarier** [comment]
This was fixed upstream early this year. There is now a full-page warning when you try to submit an insecure form:
![Screenshot from 2021-08-17 18-19-58](https://user-images.githubusercontent.com/167821/129821336-a18fb965-2c10-49fa-b1da-4a9e234c3c82.png)
